src/proto/streams/streams.rs

```diff
@@ -625,7 +680,15 @@ where
     P: Peer,
 {
     fn drop(&mut self) {
-        let mut me = self.inner.lock().unwrap();
+        let mut me = match self.inner.lock() {
+            Ok(inner) => inner,
+            Err(_) => if ::std::thread::panicking() {
+                trace!("StreamRef::drop; mutex poisoned");
```

Comment:
I noticed quite a few times when hitting assertions that this drop function was causing aborts. This code could likely stay, because aborting is 💩 .